src/js/server/dispatcher.js

```diff
@@ -21,45 +21,62 @@ var fs         = require("fs");
 var path       = require("path");
 
 gpii.express.dispatcher.route = function (that, req, res) {
-    var template = req.params.template ? req.params.template : that.options.defaultTemplate;
+    var template     = req.params.template ? req.params.template : that.options.defaultTemplate;
     var templateName = template + ".handlebars";
 
-    var viewDir          = that.options.config.express.views;
-    var templateRelPath  = path.join("pages", templateName);
-    var layoutFilename   = templateName;
-    var layoutFullPath   = path.join(viewDir, "layouts", layoutFilename);
-    if (!fs.existsSync(layoutFullPath)) {
-        layoutFilename = "main.handlebars";
-    }
+    var layoutName   = templateName;
 
-    var templateFullPath = path.join(viewDir, templateRelPath);
-    if (fs.existsSync(templateFullPath)) {
-        var contextToExpose = fluid.model.transformWithRules({ model: that.model, req: req }, that.options.rules.contextToExpose);
+    var layoutExists   = false;
+    var templateExists = false;
+    var viewDirs = fluid.makeArray(that.options.config.express.views);
+    fluid.each(viewDirs, function (viewDir) {
```

Comment:
Write this as two fluid.find loops (which you will then notice can be refactored to use a common function) rather than a single "fluid.each with side-effects".
In general, assigning to variables at an outer closure scope is a "bad smell" which should be avoided.